nixos/modules/misc/meta.nix

```diff
@@ -81,8 +139,51 @@ in
         '';
       };
 
+      tests = lib.mkOption {
+        type = testsType;
+        description = ''
+          Tests for particular modules.
+
+          Definitions are tied to the location of the module, so the definition syntax is distinct from the option value format.
+
+          Example definition:
+          ```nix
+          # module.nix
+          {
+            meta.tests = { nixosTests }:
+              {
+                inherit (nixosTests) foo;
+              };
+          }
+          ```
+
+          Example option value:
+          ```nix
+          { nixosTests }:
+          {
+            byPath = {
+              "/path/to/module.nix" = {
+                foo = nixosTests.foo;
+              };
+              # ...
+            };
+            filter = paths: { ... };
+            filterForNixBuild = paths: [ ... ];
+          }
+          ```
+
+          The filter functions have `nix repl` `:doc` documentation.
+
+          This can be used for tooling to figure out which tests are relevant, given a set of changed files.
+        '';
+        default = { ... }: { };
```

Comment:
```suggestion
        default = { ... }: { };
        defaultText = lib.literalExpression "{ ... }: { }";
```

or

```suggestion
        default = _: { };
        defaultText = lib.literalExpression "{ nixosTests }: { }";
```

Reply:
I'm un-documenting the default because it's niche and un-interesting.
The `description` covers it all.
That also means I won't add an `example` (not that you were suggesting to), as that would be confusing because the definitions and option value are so different.